third_party/move/move-compiler/src/expansion/ast.rs

```diff
@@ -414,13 +414,37 @@ pub type Type = Spanned<Type_>;
 #[derive(Debug, Clone, PartialEq)]
 pub enum LValue_ {
     Var(ModuleAccess, Option<Vec<Type>>),
-    Unpack(ModuleAccess, Option<Vec<Type>>, Fields<LValue>),
-    PositionalUnpack(ModuleAccess, Option<Vec<Type>>, LValueList),
+    Unpack(
+        ModuleAccess,
+        Option<Vec<Type>>,
+        Fields<LValue>,
+        Option<Dotdot>,
+    ),
+    PositionalUnpack(ModuleAccess, Option<Vec<Type>>, LValueOrDotdotList),
 }
 pub type LValue = Spanned<LValue_>;
 pub type LValueList_ = Vec<LValue>;
 pub type LValueList = Spanned<LValueList_>;
 
+pub fn wild_card(loc: Loc) -> LValue {
+    let wildcard = sp(loc, Symbol::from("_"));
+    let lvalue_ = LValue_::Var(sp(loc, ModuleAccess_::Name(wildcard)), None);
+    sp(loc, lvalue_)
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Dotdot_;
+pub type Dotdot = Spanned<Dotdot_>;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum LValueOrDotdot_ {
```

Comment:
Would it make sense to use `Either` instead of an explicitly named enum `XorY` (which seems like an antipattern)?

Reply:
We can avoid defining some additional structs if using `Either`. But I didn't use that because
1. I implemented `AstDebug` on `LValueOrDotdot_`, and we cannot implement that trait if using `Either`. Then I have to inline the `ast_debug` for the new`LValueOrDotdot_ = Either<...>`.
1. Matching against a `Left` or `Right` is awkward; the variant name doesn't tell anything about the variant.
3. `Either` is not official yet, and we need to introduce another dependency to `move-compiler`.